core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleWriterSuite.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort
+
+import org.mockito.Mockito._
+import org.mockito.MockitoAnnotations
+import org.scalatest.BeforeAndAfterEach
+
+import org.apache.spark.{ShuffleDependency, _}
+import org.apache.spark.memory.MemoryTestingUtils
+import org.apache.spark.serializer.JavaSerializer
+import org.apache.spark.shuffle.{BaseShuffleHandle, IndexShuffleBlockResolver}
+import org.apache.spark.util.Utils
+
+class SortShuffleWriterSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private val shuffleId = 0
+  private val numMaps = 5
+  private val conf: SparkConf = new SparkConf(loadDefaults = false)
+  private val sc = new SparkContext("local-cluster[1,1,1024]", "test", conf)
+  private var shuffleHandle: BaseShuffleHandle[Int, Int, Int] = _
+  private val shuffleBlockResolver = new IndexShuffleBlockResolver(conf)
+  private val serializer = new JavaSerializer(conf)
+  private val context = MemoryTestingUtils.fakeTaskContext(sc.env)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    MockitoAnnotations.initMocks(this)
+    val partitioner = new Partitioner() {
+      def numPartitions = numMaps
+      def getPartition(key: Any) = Utils.nonNegativeMod(key.hashCode, numPartitions)
+    }
+    shuffleHandle = {
+      val dependency = mock(classOf[ShuffleDependency[Int, Int, Int]])
+      when(dependency.partitioner).thenReturn(partitioner)
+      when(dependency.serializer).thenReturn(serializer)
+      when(dependency.aggregator).thenReturn(None)
+      when(dependency.keyOrdering).thenReturn(None)
+      new BaseShuffleHandle(shuffleId, numMaps = numMaps, dependency)
+    }
+  }
+
+  override def afterEach(): Unit = {
```

Comment:
You are creating the SparkContext for your suite member val `sc` but you are closing it after each tests.
Same for shuffleBlockResolver.
So I think you need `afterAll()` here instead of `afterEach()` and  in the finally you should call `super.afterAll()`.

Reply:
This is till suspicious for me as `shuffleBlockResolver` is initialised as a member val (and both tests uses it) but it is closed after each tests.

Reply:
the shuffleBlockResolver.stop() is actually an empty method and do nothing. But I think it's no need to call stop. Unit test IndexShuffleBlockResolverSuite also not call stop.

Reply:
I do not agree. As `IndexShuffleBlockResolver` can be closed so calling that close is our responsibility. If somebody adds some resources to this class he can rightfully assume the resource release will be handled correctly if he put it in the `IndexShuffleBlockResolver#close` method.

Anyway my suggestion is quite simple just use `afterAll()` instead of `afterEach()`. 
I have done this change locally and tested it, passed successfully:

```
[info] SortShuffleWriterSuite:
[info] - write empty iterator (103 milliseconds)
[info] - write with some records (146 milliseconds)
```